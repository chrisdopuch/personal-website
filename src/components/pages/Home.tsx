import Paper from '@material-ui/core/Paper';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import me from '../../images/me.jpg';

const stylesDeclarations = (theme: Theme) =>
  createStyles({
    '@keyframes face-spin': {
      from: { transform: 'rotate(0deg)' },
      to: { transform: 'rotate(360deg)' },
    },
    face: {
      animation: 'face-spin infinite 20s linear',
      borderRadius: '50%',
      height: '40vmin',
      margin: '10px',
      maxHeight: '300px',
      maxWidth: '300px',
      width: 'auto',
    },
    root: {
      paddingBottom: theme.spacing.unit * 2,
      paddingTop: theme.spacing.unit * 2,
      ...theme.mixins.gutters(),
    },
  });

interface IHomeProps extends WithStyles<typeof stylesDeclarations> {}

class Home extends React.Component<IHomeProps> {
  public render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.root} elevation={1}>
        <img src={me} className={classes.face} alt="my-face" />
        <Typography variant="body1">I wrote enough React to make my head spin!</Typography>
        <Typography variant="body1">More content to come soon...</Typography>
        <Typography variant="body1">
          <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
            Learn React
          </a>
        </Typography>
      </Paper>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(Home);
