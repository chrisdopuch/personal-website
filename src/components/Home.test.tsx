import { shallow } from 'enzyme';
import React from 'react';
import Home from './Home';

describe('Home', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });
});
