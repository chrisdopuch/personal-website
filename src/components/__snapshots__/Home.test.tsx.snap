// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render with default props 1`] = `
<Home
  classes={
    Object {
      "face": "Home-face-1",
    }
  }
/>
`;
